client/home.js

```diff
@@ -136,19 +138,16 @@ function handleGoToButton() {
 // eslint-disable-next-line no-unused-vars
 function handleAreaButton() {
     window.clearTimeout(currentRefreshTimer);
-    var options = [];
-
-    for (var i = 0; i < config.areas.length; i++) {
-        options.push(config.areas[i][0]);
-    }
+    /** @type {Array<string|object>} */
+    var options = zones ? zones.map(z => z.name) : [];
 
     options.push({label: "Cancel", icon: "md-close"});
 
     ons.openActionSheet({title: "Clean area", cancelable: true, buttons: options})
         .then(function(index) {
-            if (index > -1 && index < config.areas.length) {
+            if (index > -1 && zones && index < zones.length) {
```

Comment:
is the && zones && needed? Wouldn't it need a parameter?

Reply:
We need to check that zones is a valid object before checking it’s length (e.g. in case it’s null).